src/pages/user/UserBrowserPage.tsx

```diff
@@ -1,163 +1,119 @@
-import { ChevronDownIcon, CloseIcon, SearchIcon } from '@chakra-ui/icons'
+import { CloseIcon, SearchIcon } from '@chakra-ui/icons'
 import {
-  Avatar,
-  Badge,
-  Box,
-  Button,
   Flex,
   Heading,
+  IconButton,
   Input,
   InputGroup,
   InputLeftElement,
   InputRightElement,
-  Table,
-  TableCaption,
-  TableContainer,
-  Tbody,
-  Td,
-  Th,
-  Thead,
-  Tr,
-  useColorModeValue
+  useBreakpointValue,
+  useToast
 } from '@chakra-ui/react'
-import { useState } from 'react'
+import debounce from 'lodash.debounce'
+import { useEffect, useRef, useState } from 'react'
 import { Helmet } from 'react-helmet-async'
-import { useNavigate } from 'react-router-dom'
-import { PATHS } from '../../util/paths'
-import { UserPreview } from './types/UserPreview'
+import { FaChevronLeft, FaChevronRight } from 'react-icons/fa'
+import { FetchUserListMutationProps, useFecthUserListMutation } from '../../api/hooks/userMutationHooks'
+import { KonziError } from '../../api/model/error.model'
+import { generateToastParams } from '../../util/generateToastParams'
+import { ErrorPage } from '../error/ErrorPage'
+import { LoadingUserList } from './components/LoadingUserList'
+import { UserListWithPagination } from './components/UserListWithPagination'
 
-const currentUser: UserPreview = {
-  id: 1,
-  fullName: 'Kovács János',
-  presentations: 0,
-  averageRating: 0,
-  attendances: 7
-}
+export const UserBrowserPage = () => {
+  const toast = useToast()
+  const {
+    isLoading,
+    data,
+    mutate: fetchUsers,
+    error
+  } = useFecthUserListMutation((e: KonziError) => {
+    toast(generateToastParams(e))
+  })
+
+  const resultCount = useBreakpointValue({ base: 4, lg: 6, '3xl': 8 }, { ssr: false }) || 4
+  const [search, setSearch] = useState<string>('')
```

Comment:
type is redundant, because giving an initial value also determines the type. Defining which type the hook uses os only required in this situation if the value can be something else too (e.g. undefined)